pkg/uefi/firmwarevolume.go

```diff
@@ -235,6 +235,10 @@ func NewFirmwareVolume(data []byte, fvOffset uint64, resizable bool) (*FirmwareV
 	// Parse the extended header and figure out the start of data
 	fv.DataOffset = uint64(fv.HeaderLen)
 	if fv.ExtHeaderOffset != 0 && uint64(fv.ExtHeaderOffset) < fv.Length-FirmwareVolumeExtHeaderMinSize {
+		if int(fv.ExtHeaderOffset) >= len(data) {
```

Comment:
You probably don't need that test if you do the other one `if fv.Length > uint64(len(data))` first as the condition above is more restrictive : 
if fv.Length <= len(data)
and fv.ExtHeaderOffset < fv.Length-FirmwareVolumeExtHeaderMinSize
then fv.ExtHeaderOffset must be < len(data)

unless fv.Length < FirmwareVolumeExtHeaderMinSize so you might want to test for that instead as I guess it will be easier to understand the sanity check